### ISSUE TITLE
Clear search constraints when starting a new search

### ISSUE THREAD

**Issue by dylandforbes** [arybczak/ncmpcpp#221]
Not sure why this has started happening since 0.8. Hitting "/", entering a string, hittine either enter or Ctrl-C, then hitting "/" again leaves the search term as it was before, requiring one to hit backspace over and over before doing a new search.

**Comment by hghwng** [comment]
I don't think it's an issue here.

If you want to edit your previous search, in this case, filling the blank with the last executed search is the desired behavior.

If you want to do another search, you can easily clear the input box by pressing Ctrl + U. The input box of ncmpcpp  is provided by readline, so there are many [shortcuts](http://readline.kablamo.org/emacs.html) can accelerate your edit.

**Comment by ayekat** [comment]
> If you want to do another search, you can easily clear the input box by pressing Ctrl + U. The input box of ncmpcpp is provided by readline, so there are many [shortcuts](http://readline.kablamo.org/emacs.html) can accelerate your edit.

While `Ctrl`+`U` *does* work, `Ctrl`+`P` and `Ctrl`+`N` (for selecting the previous/next entry in the history) don't do anything. Or is that just me?

**Comment by hghwng** [comment]
@ayekat It seems that the history is kept in memory only. I can scroll though history items inside one ncmpcpp session. After relaunching it, the history is gone.

**Comment by seeegma** [comment]
Readline is all fine and good. For me it is simply a *major* hindrance when searching. I never want to search for the same thing twice, so even hitting Ctrl-U is quite annoying. This could certainly be made a toggleable option.

**Comment by seeegma** [comment]
@arybczak What is the story behind this change?

**Comment by arybczak** [comment]
It was made for consistency with filtering. When you set a filter and press Ctrl+F again, you'll have the filter shown, so the same was done with searching. However, it was also slightly annoying to me in the beginning, so perhaps it's best to restore previous behavior.

**Comment by nonamethanks** [comment]
+1. Having the search not reset is also useless as pressing enter does not select the next result.

**Comment by seeegma** [comment]
@arybczak Ah, I see. It certainly makes sense to bring up the current filter string when hitting Ctrl-F, because the filter is determined just by what the user has typed—they don't need to hit enter to 'activate' the filter. Thus, I don't think it makes sense to match this behavior with searching. Bringing up a new search doesn't alter the active search string; you need to hit enter.

**Comment by djvs** [comment]
Agree, this is a regression...

**Comment by djvs** [comment]
A config option maybe?

**Comment by NicolasCARPi** [comment]
Hello,

Please add a config option to clear input for new searches :)

Relevant xkcd:
![workflow](https://imgs.xkcd.com/comics/workflow.png)

**Comment by rubyconn** [comment]
I hope removal of this feature is being considered, or at least a config option to disable it.

**Comment by NicolasCARPi** [comment]
Yeah it's beginning to be really annoying… When using mutt: / is clear, vim: / is clear, so why ncmpcpp annoys me with the previous search?

**Comment by rubyconn** [comment]
I added a PR. Until the changes are commited you can use the following binding definitions to achieve the same behavior.

```
def_key "/"
  push_character "ctrl-u"
  find_item_forward

def_key "?"
  push_character "ctrl-u"
  find_item_backward
```

Thanks to GhostValk in #228 for figuring this out!

**Comment by arybczak** [comment]
Fixed by fa9838efb28cb17a81a6dde07d91233a2260c4f5
